lib/Site.js

```diff
@@ -74,6 +77,37 @@ function setExtension(filename, ext) {
   );
 }
 
+/**
+* Creates a function that delays invoking `func` until after `wait` milliseconds have elapsed
+* and the running `func` has resolved/rejected.
+* @param func the promise-returning function to delay
+* @param wait the number of milliseconds to delay
+*/
+function delay(func, wait) {
+  let context;
+  let pendingArgs = [];
+  let waitingPromise = null;
+  let runningPromise = Promise.resolve();
+
+  return function (arg) {
+    context = this;
+    pendingArgs.push(arg);
```

Comment:
Also, for the non-prefixed method e.g. `rebuildSourceFiles`, 
it should only take in one `filePath` parameter.
Currently, that is how it is used in `index.js` too.

If it is supposed to be multiple parameters
then the function should take in `...args`, we should `concat` instead of `push` args or flatten the array in the delayed functions

Reply:
As I commented above: *The way before 55b874a is preferable*.
The spread syntax allowed the "wrapped" function to take in one or more parameters.
We should use `concat` since the "original" function accepted an array (seems clean).